Assets/World/Track/Scripts/RandomTrackGenerator.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+/// <summary>
+/// A track generator that randomly generates your tracks.
+/// </summary>
+public class RandomTrackGenerator : TrackGeneratorCommon
+{
+    [SerializeField]
+    GameObject m_firstTrackPiece; // Temporary, we will programatically generate the first track piece in the future.
+
+    /// <summary>
+    /// Generate tracks by getting the first track piece, then grabbing a random track piece from resources and joining
+    /// it together. Track pieces are moved and rotated to the position of the 'Track Piece Link' on the previous Track Piece.
+    /// </summary>
+    /// <param name="trackLength"></param>
+    /// <param name="trackAltitude"></param>
+    /// <param name="availableTrackPiecePrefabs"></param>
```

Comment:
```suggestion
     /// <param name="availableTrackPiecePrefabs"></param>
     /// <param name="availableTrackPiecePrefabs">Description of the Param</param>
```

Need a brief description of the param docs, otherwise they're useless